container/src/test/java/com/venta/container/cyclic/ComponentB.java

```diff
@@ -0,0 +1,17 @@
+package com.venta.container.cyclic;
+
+import io.github.trimax.venta.container.AbstractVentaApplication;
+import io.github.trimax.venta.container.annotations.Component;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Component
+public class ComponentB implements AbstractVentaApplication<Void> {
+    public ComponentB(ComponentA a) {
+    }
```

Comment:
No need. Just add ComponentA as a private final field and define AllArgs with lombok